pkg/protocol/engine/congestioncontrol/scheduler/drr/scheduler.go

```diff
@@ -337,15 +328,66 @@ func (s *Scheduler) selectIssuer(start *IssuerQueue) (int64, *IssuerQueue) {
 	return rounds, schedulingIssuer
 }
 
+func (s *Scheduler) removeIssuer(q *IssuerQueue) {
+	q.submitted.ForEach(func(id iotago.BlockID, block *blocks.Block) bool {
+		block.SetDropped()
+		s.events.BlockDropped.Trigger(block)
+
+		return true
+	})
+
+	for q.inbox.Len() > 0 {
+		block := q.PopFront()
+		block.SetDropped()
+		s.events.BlockDropped.Trigger(block)
+	}
+
+	s.buffer.RemoveIssuer(q.IssuerID())
+}
+
+func (s *Scheduler) getDeficit(accountID iotago.AccountID) (uint64, error) {
+	d, exists := s.deficits.Get(accountID)
+	if !exists {
+		_, err := s.manaRetrieveFunc(accountID)
+		// manaRetrieveFunc return error if mana is less than MinMana or the Mana can not be retrieved for the account.
+		if err != nil {
+			return 0, ierrors.Wrapf(err, "could not get deficit for issuer %s", accountID)
+		}
+		// load with max deficit if the issuer has Mana but has been removed from the deficits map
+		return s.optsMaxDeficit, nil
+	}
+
+	return d, nil
+}
+
 func (s *Scheduler) updateDeficit(accountID iotago.AccountID, delta int64) error {
-	deficit, err := s.getDeficit(accountID)
+	var err error
+	s.deficits.Compute(accountID, func(currentValue uint64, exists bool) uint64 {
+		if !exists {
+			_, err = s.manaRetrieveFunc(accountID)
+			// manaRetrieveFunc return error if mana is less than MinMana or the Mana can not be retrieved for the account.
+			if err != nil {
+				err = ierrors.Wrapf(err, "could not get deficit for issuer %s", accountID)
+				return 0
+			}
+
+			// load with max deficit if the issuer has Mana but has been removed from the deficits map
+			return s.optsMaxDeficit
+		}
+
+		if int64(currentValue)+delta < 0 {
```

Comment:
Shouldn't we use the safemath functions?

Reply:
Yeah! But I think applying safemath belongs to a separate PR that changes this in all the places across codebase. We're not using that anywhere yet, right?

Reply:
Sure, can you add a TODO there so we don't miss it?